src/sidebar/store/modules/annotations.ts

```diff
@@ -567,6 +573,37 @@ const savedAnnotations = createSelector(
   annotations => annotations.filter(ann => isSaved(ann)) as SavedAnnotation[],
 );
 
+/**
+ * Return the list of unique users who authored any annotation, ordered by username.
+ */
+const usersWhoAnnotated = createSelector(
+  (state: State) => state.annotations,
+  annotations => {
+    const usersMap = new Map<
+      string,
+      { user: string; username: string; displayName: string | null }
+    >();
+    annotations.forEach(anno => {
+      const { user } = anno;
+      const username = getUsername(user);
+      const displayName = anno.user_info?.display_name ?? null;
+
+      // Keep a unique list of users
+      if (!usersMap.has(user)) {
```

Comment:
This check could be moved up so you bail out earlier if the username has already been seen and skip extracting the username and display name in that case.